### ISSUE TITLE
By bullet caliber minimum damage allowance   enhancement/refinement

### ISSUE THREAD

**Issue by SpannerMonkey** [PapaJoesSoup/BDArmory#515]
Enhancement
##### KSP and BDA Version
All post damage overhaul

##### OS / ENVIRONMENT
All

##### SUMMARY
Low caliber weapons do little to no appreciable damage, for example 7mm bullet using standard BDA bullet def  is  removing  approx 0.025 HP per shot on average , to an unarmored target 
The situation continues right through to 120 mm, as used by many tank and marine turrets. 

It can be easily seen that currently the ultimate caliber/weapon for normal user combat is a 30mm rapid fire turret, any one of these is capable of reducing any normal player vehicle to wreckage in seconds. 
While anything below 30mm does little to no damage , and larger calibers in particular non explosive types, do only a fraction of the damage over time .

There is an imbalance between the damage caused by explosive and penetrating shells,  that currently has only two solutions,  for anyone wanting not to use 30mm guns,  and they are,  for low calibers,  to alter the bullet caliber mass and velocity to fantasy figures, ( In one case , a supposedly 7mm gun firing 20mm bullets)   and for larger calibers to switch to the original BDA everything explodes method  with HE for everything.

I feel the system is mature enough now to stand a bit of criticism, and I think that most weapon mod makers would agree with me, that while the old heat based system was less than perfect it made for an ultimately better balanced(for the general user) game play and range of weapons,   and much easier to make weapons more or less effective as required. 
  I recall that the driving force behind the changes was to create a better playing experience and a more balanced playing field, and currently I and many others  feel that it has yet to get there.

I'm a big supporter of the damage overhaul in general, as it suits huge craft and weapons very well.  However it is important that the mod performs well for all types of player,  from those that use  primitive 7mm guns on lightweight aircraft , to those who build several thousand tonne ships with 16" guns 

Do note that I'm very familiar with all the cfg options for damage multiplier etc , however these do not make up for the ineffective low calibers ( even when set at ludicrous levels) or the low damage over time of non explosive shells. 

The question ultimately is, can anything be done to sort out the current horrible balance between calibers and damage output?

##### STEPS TO REPRODUCE
Try it yourself, objectively

**Comment by NextStarIndustries** [comment]
So fix it! Aren't you a BDA god?! LMAO

**Comment by sjpc302** [comment]
Agree 100%, I only was able to get my light ap cannon to a satisfactory state by changing the ballistic multiplier to 3 and then adding a bullet dmg multiplier of 3 to the gun. With only mg's I found it was more effective to ram the enemy planes and pray

**Comment by gomker** [comment]
So let me start with the "why" - The damage system is based on real physics - basically F=MA which is measured in Newtons. If you run these numbers for the velocity of a bullet you get HUGE numbers. I need to reconcile these numbers with more realistic game play numbers for Hitpoints otherwise HP ends up being ludicrous (say 1 million+) which is not easy to understand. So I scaled it

            float damage_ = ((0.5f * (mass * Mathf.Pow(impactVelocity, 2)))
                            * (BDArmorySettings.DMG_MULTIPLIER / 100) * bulletDmgMult
                            * 1e-4f);

This works for most things but kills the smaller numbers to stupid low figures. This is why I added the multiplier. This multiplier is currently a straight number. I can hardcode another factor multiply it by to it in code or add another value in BDA Settings to let you guys figure out what is right.  I can code it so those values only apply if set in the config so you wont be messing with other bullet numbers already established.

Basically I want to give you guys as much control over it as possible
